sbysrc/sby_core.py

```diff
@@ -307,23 +311,28 @@ def parse_config(self, f):
                     self.autotune_config = sby_autotune.SbyAutotuneConfig()
                     continue
 
-                if entries[0] == "file":
+                if section == "file":
                     mode = "file"
-                    if len(entries) != 2:
+                    if args is None:
                         self.error(f"sby file syntax error: '[file]' section expects a file name argument")
-                    current_verbatim_file = entries[1]
+
+                    section_args = args.split()
+
+                    if len(section_args) > 1:
+                        self.error(f"sby file syntax error: '[file]' section expects exactly one file name argument, got {len(section_args)}")
+                    current_verbatim_file = section_args[0]
                     if current_verbatim_file in self.verbatim_files:
-                        self.error(f"duplicate file: {entries[1]}")
+                        self.error(f"duplicate file: {current_verbatim_file}")
                     self.verbatim_files[current_verbatim_file] = list()
                     continue
 
-                if entries[0] == "files":
+                if section == "files":
                     mode = "files"
-                    if len(entries) != 1:
-                        self.error(f"sby file syntax error: '[files]' section expects no arguments, got '{line}'")
+                    if args is not None:
+                        self.error(f"sby file syntax error: '[files]' section does not accept any arguments. got {args}")
                     continue
 
-                self.error(f"sby file syntax error: unexpected section '{entries[0]}', expected one of 'options, engines, script, autotune, file, files'")
+                self.error(f"sby file syntax error: unexpected section '{section}', expected one of 'options, engines, script, autotune, file, files'")
 
             if mode == "options":
                 entries = line.split()
```

Comment:
This was the split we were talking about yesterday, as this is where `expect pass, fail` currently fails with an error message that complains about a missing argument, but it's the same situation / code as for the section headers.

Reply:
Sorry, this comment was supposed to be a bit further down _on_ line 329/338 but it seems github wants to focus on the nearest change in the commit instead.

Reply:
Ah, right, I just changed how the section headers were parsed at the moment not the lines belonging to the sections yet, I suppose I misunderstood what was related to that, my bad.

Would it be worth putting the line argument parsing in a separate PR or joined up with this one?

Also slightly unrelated, what are your thoughts on breaking the `SbyConfig` out into its own file to make maintenance a little easier? I know for me I keep getting lost in the file.

Reply:
Both, separate or same PR, is fine with me. Breaking out `SbyConfig` sounds like a good idea to me.

Reply:
@jix I'll break out the `SbyConfig` into its own module in another PR, but for now but other than the minor fix for the entries in options anything else?